File: two_pointers/two_sum_ii_input_array_is_sorted/two_sum.py
def two_sum(nums, target):
    l, r = 0, len(nums) - 1

    while l < r:
        l_value = nums[l]
        r_value = nums[r]
        if l_value + r_value == target:
            return [l + 1, r + 1]
        if l_value + r_value > target:
            r -= 1
            continue
        l += 1
    return []

File: two_pointers/two_sum_ii_input_array_is_sorted/two_sum.go
// Package twosum will solve the Two Integer Sum II problem.
package twosum

func twoSum(nums []int, target int) []int {
	leftPointer, rightPointer := 0, len(nums)-1
	for leftPointer < rightPointer {
		leftValue := nums[leftPointer]
		rightValue := nums[rightPointer]
		if leftValue+rightValue == target {
			return []int{leftPointer + 1, rightPointer + 1}
		}
		if leftValue+rightValue > target {
			rightPointer--
			continue
		}
		leftPointer++
	}
	return []int{}
}
